extensions/amp-moviads-player/0.1/amp-moviads-player.js

```diff
@@ -0,0 +1,890 @@
+        element,
+        'ima-video',
+        {initialConsentState},
+        {allowFullscreen: true}
+      );
+
+      this.applyFillContent(iframe);
+
+      // This is temporary until M74 launches.
+      // TODO(aghassemi, #21247)
+      addUnsafeAllowAutoplay(iframe);
+
+      this.iframe_ = iframe;
+
+      const deferred = new Deferred();
+      this.playerReadyPromise_ = deferred.promise;
+      this.playerReadyResolver_ = deferred.resolve;
+
+      this.unlistenMessage_ = listen(this.win, 'message', e =>
+        this.handlePlayerMessage_(/** @type {!Event} */ (e))
+      );
+
+      element.appendChild(iframe);
+
+      installVideoManagerForDoc(element);
+      Services.videoManagerForDoc(element).register(this);
+
+      return this.loadPromise(iframe).then(() => this.playerReadyPromise_);
+    });
+  }
+
+  /** @override */
+  viewportCallback(visible) {
+    this.element.dispatchCustomEvent(VideoEvents.VISIBILITY, {visible});
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    if (this.unlistenMessage_) {
+      this.unlistenMessage_();
+    }
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+    return true;
+  }
+
+  /** @override */
+  onLayoutMeasure() {
+    if (!this.iframe_) {
+      return;
+    }
+    const {width, height} = this.getLayoutBox();
+    this.sendCommand_('resize', {'width': width, 'height': height});
+  }
+
+  /**
+   * Sends a command to the player through postMessage. NOTE: All commands sent
+   * before imaVideo fires VideoEvents.LOAD will be queued until that event
+   * fires.
+   * @param {string} command
+   * @param {Object=} opt_args
+   * @private
+   */
+  sendCommand_(command, opt_args) {
+    if (this.iframe_ && this.iframe_.contentWindow) {
+      this.playerReadyPromise_.then(() => {
+        this.iframe_.contentWindow./*OK*/ postMessage(
+          JSON.stringify(
+            dict({
+              'event': 'command',
+              'func': command,
+              'args': opt_args || '',
+            })
+          ),
+          '*'
+        );
+      });
+    }
+    // If we have an unlistener for this command, call it.
+    if (this.unlisteners_[command]) {
+      this.unlisteners_[command]();
+    }
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  handlePlayerMessage_(event) {
+    if (event.source != this.iframe_.contentWindow) {
+      return;
+    }
+    const eventData = getData(event);
+    if (!isObject(eventData)) {
+      return;
+    }
+
+    const videoEvent = eventData['event'];
+
+    if (videoEvent === VideoEvents.AD_START) {
+      this.sendCommand_('gaParams', {
+        UA: this.gaUserUA_,
+        name: this.gaUser_,
+        id: this.config_.movieContenerId,
+      });
+
+      this.isAdStart_ = true;
+      this.eventGA('AD_START', this.gaUser_, this.gaUserUA_);
```

Comment:
Absolutely not. Remove `eventGA`. Use [video analytics support.](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-video-analytics.md)